flink-libraries/flink-table/src/main/scala/org/apache/flink/table/runtime/aggregate/AggregateUtil.scala

```diff
@@ -1393,6 +1393,21 @@ object AggregateUtil {
             throw new TableException(s"unsupported Function: '${unSupported.getName}'")
         }
       }
+
+      // create distinct accumulator delegate
+      if (aggregateCall.isDistinct) {
```

Comment:
It is better to move this logic into the upper match. Distinct is also a kind of aggregate whose fields should not be empty. Also, we can reuse some variables. What do you think?

Reply:
Good point. Actually there's a very interesting question I've been thinking - This AggFunction literally is the distinct version of the  `CollectAggFunction`, however it cannot exist by itself unless chained with other `realAgg` functions --> this means we will need to chain `DistinctAggFunction` with `CollectAggFunction` which just dont make sense. I will try some other approach and see if they work.

Reply:
Sql will be verified by calcite to exclude single DISTINCT during sql parse phase, so maybe we don't have to consider single distinct here.